File: libs/analytics/src/main/java/com/example/analytics/AnalyticsActivityLifecycle.java
package com.example.analytics;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

public class AnalyticsActivityLifecycle implements Application.ActivityLifecycleCallbacks {
  @Override
  public void onActivityCreated(Activity activity, Bundle bundle) {

  }

  @Override
  public void onActivityStarted(Activity activity) {

  }

  @Override
  public void onActivityResumed(Activity activity) {
    if (activity instanceof TraceableScreen) {
      AnalyticsUtils.trackView(activity, ((TraceableScreen) activity).getScreenName());
    }
  }

  @Override
  public void onActivityPaused(Activity activity) {

  }

  @Override
  public void onActivityStopped(Activity activity) {

  }

  @Override
  public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {

  }

  @Override
  public void onActivityDestroyed(Activity activity) {

  }
}

File: libs/analytics/src/main/kotlin/com/example/analytics/AnalyticsActivityLifecycle.kt
package com.example.analytics

import android.app.Activity
import android.app.Application
import android.os.Bundle

class AnalyticsActivityLifecycle : Application.ActivityLifecycleCallbacks {
  override fun onActivityCreated(activity: Activity, bundle: Bundle) {
  }

  override fun onActivityStarted(activity: Activity) {
  }

  override fun onActivityResumed(activity: Activity) {
    if (activity is TraceableScreen) {
      AnalyticsUtils.trackView(activity, (activity as TraceableScreen).screenName)
    }
  }

  override fun onActivityPaused(activity: Activity) {
  }

  override fun onActivityStopped(activity: Activity) {
  }

  override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {
  }

  override fun onActivityDestroyed(activity: Activity) {
  }
}
